numba_dpex/core/kernel_interface/dispatcher.py

```diff
@@ -57,6 +59,7 @@ def __init__(
         debug_flags=None,
         compile_flags=None,
         array_access_specifiers=None,
+        enable_cache=True,
```

Comment:
Why are we updating this function? The function is a way to keep track of an optional "array access specifier" (readonly, write, read-write) for a NumPy array passed to a kernel as an argument. 

The `get_ordered_arg_access_types` function should be unrelated to the kernel caching mechanism. Moreover, I am going to remove the function once we remove NumPy argument support for kernels.